module/zfs/zfs_ioctl.c

```diff
@@ -5821,6 +5822,44 @@ zfs_ioc_send_space(const char *snapname, nvlist_t *innvl, nvlist_t *outnvl)
 	return (error);
 }
 
+/*
+ * Sync the currently open TXG to disk for the specified pool.
+ * This is somewhat simular to 'zfs_sync()'.
+ * For all cases that do not result in error this iotcl will wait for
+ * the currently open TXG to commit before returning back to the caller.
+ *
+ * innvl is unused
+ *
+ * onvl is unused
+ */
+/* ARGSUSED */
+static int
+zfs_ioc_pool_sync(const char *pool, nvlist_t *innvl, nvlist_t *onvl)
+{
+	int err;
+	spa_t *spa;
+
+	ASSERT3P(innvl, ==, NULL);
+
+	/* we don't want a panicking kernel writing to storage */
+	if (unlikely(oops_in_progress))
+		return (EHOSTDOWN);
```

Comment:
I don't think this is necessary.  This function isn't doing anything but waiting for the txg to sync which lots of the existing ioctl's do.  Is the intention to avoid getting blocked here in the case of a panic?